apache/couchdb
# Enable discovery of cluster nodes through DNS SRV

## Overview

This patch enables CouchDB nodes to check SRV records in DNS for cluster peers and automatically add any discovered peers to the nodes DB. This behavior is disabled by default and can be enabled as follows:

[mem3]
use_dns_service_discovery = true
service_record_name = _couchdb._tcp.example.net

If the `service_record_name` is omitted we attempt to generate the appropriate record for the query by prepending _couchdb._tcp. to the DNS domain name portion of the host / VM / container FQDN.

I wrote this code a few months back and it passed some smoke tests in Kube but I never opened a PR. Opening it now so I don't forget about it.

## Testing recommendations

1) Start one CouchDB node
2) Create an SRV record pointing to it
3) Start a second node with the [mem3] configuration above
4) Check to see if they automatically join into a cluster

## Related Issues or Pull Requests

Fixes #1108 

## Checklist
